Error:  compiling  example List.ino esp32

### Discussed in https://github.com/forntoh/LcdMenu/discussions/99

<div type='discussions-op-text'>

<sup>Originally posted by **maansch** June 14, 2023</sup>
Hi Thomas,
try to use the List.ino, during compiling I receive these error msg.

/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h: In constructor 'ItemList::ItemList(const char*, String*, uint8_t, fptrInt)':
/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h:34:21: error: 'ItemList::items' will be initialized after [-Werror=reorder]
     String* items = NULL;     ///< Pointer to an array of items
                     ^
/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h:33:24: error:   'void (* ItemList::callback)(uint8_t)' [-Werror=reorder]
     fptrInt callback = NULL;  ///< Pointer to a callback function
                        ^
In file included from /tmp/arduino_modified_sketch_881776/List.ino:21:0:
/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h:47:5: error:   when initialized here [-Werror=reorder]
     ItemList(const char* key, String* items, const uint8_t itemCount,
     ^
cc1plus: some warnings being treated as errors

Bibliothek LcdMenu in Version 3.0.1 im Ordner: /home/mas/A_Prog_A/Arduino/libraries/LcdMenu  wird verwendet
Bibliothek LiquidCrystal_I2C in Version 1.1.4 im Ordner: /home/mas/A_Prog_A/Arduino/libraries/LiquidCrystal_I2C  wird verwendet
Bibliothek Wire in Version 1.0.1 im Ordner: /home/mas/arduinoide/arduino-1.8.19_ESP32/portable/packages/esp32/hardware/esp32/1.0.6/libraries/Wire  wird verwendet

exit status 1
Fehler beim Kompilieren für das Board ESP32 Dev Module.

Do you have an idea where the error could be?
I would be grateful for your help.

br
Marc-Andre


Code: example List.ino
environment : Arduino: 1.8.19 (Linux),
Board Typ: esp32
AZ-Delivery ESP-32Development-Board, ESP-WROOM-32 Prozessor 160MHz, Tensilica L108 32 bit Dual-Core CPU,  512 KB SRAM & 16 MB Flashspeicher.
Bord driver for testing:
Core Version 2.9.0 
Core Version 1.6.0 </div>

--- a/src/ItemList.h
+++ b/src/ItemList.h
@@ -31,8 +31,8 @@ class ItemList : public MenuItem {
    private:
     fptrInt callback = NULL;  ///< Pointer to a callback function
     String* items = NULL;     ///< Pointer to an array of items
-    uint8_t itemIndex = 0;    ///< The current selected item index
     const uint8_t itemCount;  ///< The total number of items in the list
+    uint8_t itemIndex = 0;    ///< The current selected item index
 
    public:
     /**
@@ -46,10 +46,10 @@ class ItemList : public MenuItem {
      */
     ItemList(const char* key, String* items, const uint8_t itemCount,
              fptrInt callback)
-        : MenuItem(key, MENU_ITEM_LIST),
-          itemCount(itemCount),
-          callback(callback),
-          items(items) {}
+        : MenuItem(key, MENU_ITEM_LIST), itemCount(itemCount) {
+        this->callback = callback;
+        this->items = items;
+    }
 
     /**
      * @brief Returns the index of the currently selected item.

